### PR TITLE
Enhancements and Updates Across Inventory and Sales Packages

### LINKED ISSUE #86: sale - cron controller to generate subscription renewals

### PR DESCRIPTION
**Pull request by ghost**
- Add expired alert in inventory package.
- Add 'is_auto_renew' field in the Subscription in the sale package.
- Add cron controller to generate subscription renewals in the sale package.
- Update order fields in advanced search in the inventory package.

### CODE REVIEW

File: inventory/actions/service/bulk-check-expirations.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    'response' => [
+        'content-type'      => 'application/json',
+        'charset'           => 'utf-8',
+        'accept-origin'     => '*'
+    ],
+    'providers' => ['context']
+]);
+
+list($context) = [$providers['context']];
```
Comment (lucaslaurent04) [inline inventory/actions/service/bulk-check-expirations.php]: could be:

```php
$provider = $providers['context'];
```

File: inventory/actions/service/bulk-check-expirations.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (lucaslaurent04) [inline inventory/actions/service/bulk-check-expirations.php]: Maybe throw invalid param error if $params['ids'] is empty

File: inventory/actions/service/check-expiration.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+$subscription = Subscription::id($subscription_id)
+    ->read([
+        'id',
+        'is_expired',
+        'has_upcoming_expiry',
+    ])
+    ->first(true);
+
+$result = [];
```
Comment (lucaslaurent04) [inline inventory/actions/service/check-expiration.php:41]: Don't think you need a $result

**Comment by lucaslaurent04** [comment]
Maybe the file packages/inventory/init/data/inventory_core_Task.json should be moved to the sale package.

And maybe you could replace the Task in this file by your new action to add the alerts ?

**Comment by ghost** [comment]
> Maybe the file packages/inventory/init/data/inventory_core_Task.json should be moved to the sale package.
> 
> And maybe you could replace the Task in this file by your new action to add the alerts ?

A bulk expiration action has been created that updates and checks the expirations of subscriptions in the inventory and sales package. This action runs automatically every day but can also be triggered manually from the subscriptions list.

Also, the code suggestions have been incorporated.

**Pull request merged**